### PR TITLE
Add scheduler example

### PR DESCRIPTION
**Pull request by yipliu**
## What does this PR do?

Add scheduler example related issue in #433 

## Before submitting

- [x] Did you make sure **title is self-explanatory** and **the description concisely explains the PR**?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **test your PR locally** with `pytest` command?
- [x] Did you **run pre-commit hooks** with `pre-commit run -a` command?

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: configs/model/mnist.yaml [hunk 1]
```diff
@@ -6,6 +6,13 @@ optimizer:
   lr: 0.001
   weight_decay: 0.0
 
+scheduler:
+  _target_: torch.optim.lr_scheduler.ReduceLROnPlateau
+  _partial_: true
+  mode: min
```
Comment (ashleve) [inline configs/model/mnist.yaml:12]: this should be set to "min" for loss, and "max" if we're tracking accuracy

File: src/models/mnist_module.py [hunk 2]
```diff
@@ -122,8 +123,17 @@ def configure_optimizers(self):
...
         """
+        optimizer = self.hparams.optimizer(params=self.parameters())
+        scheduler = self.hparams.scheduler(optimizer=optimizer)
+
         return {
-            "optimizer": self.hparams.optimizer(params=self.parameters()),
+            "optimizer": optimizer,
+            "lr_scheduler": {
+                "scheduler": scheduler,
+                "monitor": "val/acc_best",
```
Comment (ashleve) [inline src/models/mnist_module.py:133]: I think schedulers are usually optimized on validation loss.

Accuracy can remain the same for a longer time even though loss is improving, so accuracy breakthrough might get delayed

File: configs/trainer/default.yaml [hunk 3]
```diff
@@ -11,6 +11,9 @@ devices: 1
 # mixed precision for extra speed-up
 # precision: 16
 
+# Perform a validation loop every every N training epochs
+check_val_every_n_epoch: 1
+
```
Comment (ashleve) [inline configs/trainer/default.yaml:16]: what is the reason for adding this config option?

**Review by ashleve — CHANGES_REQUESTED** [review]

File: src/models/mnist_module.py (hunk 2 above)
Comment (ashleve) [inline src/models/mnist_module.py:133]:
```suggestion
                "monitor": "val/loss",
```

File: configs/trainer/default.yaml (hunk 3 above)
Comment (ashleve) [reply configs/trainer/default.yaml:16]: okay, this makes sense

File: configs/trainer/default.yaml [hunk 4]
```diff
@@ -11,6 +11,9 @@ devices: 1
 # mixed precision for extra speed-up
 # precision: 16
 
+# Perform a validation loop every every N training epochs
```
Comment (ashleve) [inline configs/trainer/default.yaml:14]:
```suggestion
# perform a validation loop every N training epochs
```

**Review by ashleve — CHANGES_REQUESTED** [review]

**Review by ashleve — APPROVED** [review]

**Comment by ashleve** [comment]
@yipliu Thank you for the contribution! I merged the changes.
